src/api.rs

```diff
@@ -1133,6 +1140,506 @@ impl<T: Pixel> ContextInner<T> {
     while self.set_frame_properties(self.next_lookahead_output_frameno).is_ok()
     {
       self.next_lookahead_output_frameno += 1;
+
+      // Compute the lookahead motion vectors.
```

Comment:
I'm concerned by the size of this function now. Is it possible to break this into smaller functions?